server/session/TransactionOLTP.java

```diff
@@ -213,31 +216,43 @@ private void persistInternal() throws InvalidKBException {
         session.keyspaceStatistics().commit(this, uncomittedStatisticsDelta);
         LOG.trace("Graph is valid. Committing graph...");
         janusTransaction.commit();
-        String txId = this.janusTransaction.toString();
+        ackCommit();
+        LOG.trace("Graph committed.");
+    }
 
-        Set<String> newAttributeIndices = cache().getNewAttributes().keySet().stream().map(Pair::second).collect(Collectors.toSet());
+    private void ackCommit(){
+        String txId = this.janusTransaction.toString();
         session.shardManager().ackCommit(cache().getNewShards().keySet(), txId);
-        session.attributeManager().ackCommit(newAttributeIndices, txId);
+        //this should ack all inserts so that insert requests are cleared
+        session.attributeManager().ackCommit(cache().getNewAttributes().keySet(), txId);
 
-        LOG.trace("Graph committed.");
+        //this should ack all inserts that weren't deduplicated so that we have correct attributes in attributesCommitted
```

Comment:
I understand this, but if we are merging 2 attributes A, it's because A was already in `attributesCommitted`, right?
So would it be so bad to re-put?
We could safe a field and some additional logic, but maybe I am missing something. I keep getting confused

Reply:
if we add all new attributes, we will overwrite legit entries in `attributesCommitted` with the ones that were deduplciated